RecoHI/HiTracking/python/hiMixedTripletStep_cff.py

```diff
@@ -1,206 +1,159 @@
+         maxChi2 = cms.double(9.0),
+     ),
+     Strip = cms.PSet(
+        #Yen-Jie's mod to preserve merged clusters
+        maxSize = cms.uint32(2),
+        maxChi2 = cms.double(9.0)
+     )
+)
 
 # SEEDING LAYERS
-from RecoLocalTracker.SiStripClusterizer.SiStripClusterChargeCut_cfi import *
-hiMixedTripletSeedLayersA = cms.EDProducer("SeedingLayersEDProducer",
-                                             layerList = cms.vstring('FPix1_pos+FPix2_pos+TEC1_pos', 'FPix1_neg+FPix2_neg+TEC1_neg'),
-                                                                     #'FPix2_pos+TEC2_pos+TEC3_pos', 'FPix2_neg+TEC2_neg+TEC3_neg'),
-                                   BPix = cms.PSet(
-    TTRHBuilder = cms.string('TTRHBuilderWithoutAngle4MixedTriplets'),
-    HitProducer = cms.string('siPixelRecHits'),
-    skipClusters = cms.InputTag('hiMixedTripletClusters')
-    ),
-                                   FPix = cms.PSet(
-    TTRHBuilder = cms.string('TTRHBuilderWithoutAngle4MixedTriplets'),
-    HitProducer = cms.string('siPixelRecHits'),
-    skipClusters = cms.InputTag('hiMixedTripletClusters')
-    ),
-                                   TEC = cms.PSet(
-    matchedRecHits = cms.InputTag("siStripMatchedRecHits","matchedRecHit"),
-    useRingSlector = cms.bool(True),
-    TTRHBuilder = cms.string('WithTrackAngle'),
-    clusterChargeCut = cms.PSet(refToPSet_ = cms.string('SiStripClusterChargeCutNone')),
-    minRing = cms.int32(1),
-    maxRing = cms.int32(1),
-    skipClusters = cms.InputTag('hiMixedTripletClusters')
-    )
-                                   )
-
-# SEEDS
-from RecoPixelVertexing.PixelTriplets.PixelTripletLargeTipGenerator_cfi import *
-PixelTripletLargeTipGenerator.extraHitRZtolerance = 0.0
-PixelTripletLargeTipGenerator.extraHitRPhitolerance = 0.0
-import RecoTracker.TkSeedGenerator.GlobalSeedsFromTriplets_cff
-hiMixedTripletSeedsA = RecoTracker.TkSeedGenerator.GlobalSeedsFromTriplets_cff.globalSeedsFromTriplets.clone()
-hiMixedTripletSeedsA.OrderedHitsFactoryPSet.SeedingLayers = 'hiMixedTripletSeedLayersA'
-hiMixedTripletSeedsA.OrderedHitsFactoryPSet.GeneratorPSet = cms.PSet(PixelTripletLargeTipGenerator)
-hiMixedTripletSeedsA.SeedCreatorPSet.ComponentName = 'SeedFromConsecutiveHitsTripletOnlyCreator'
-hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.ptMin = 4.0
-hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.originRadius = 0.005
-#hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.nSigmaZ = 4.0
-hiMixedTripletSeedsA.RegionFactoryPSet.RegionPSet.originHalfLength = 10.0
-
-hiMixedTripletSeedsA.OrderedHitsFactoryPSet.GeneratorPSet.maxElement = 5000000
-hiMixedTripletSeedsA.ClusterCheckPSet.MaxNumberOfPixelClusters = 5000000
-hiMixedTripletSeedsA.ClusterCheckPSet.MaxNumberOfCosmicClusters = 50000000
+mixedTripletStepSeedLayersA.layerList = cms.vstring('BPix1+BPix2+BPix3', 'BPix1+BPix2+FPix1_pos', 'BPix1+BPix2+FPix1_neg', 
+                                        'BPix1+FPix1_pos+FPix2_pos', 'BPix1+FPix1_neg+FPix2_neg', 
```

Comment:
are these really not picked up in the previous triplet iterations?

Reply:
Although the layer list is the same as previous iterations, the size of the tracking region is much larger.  On line 43 we are using a fixed halfLength of nearly 4 cm, while the previous triplet steps use 0.5 cm (high pt step) and less than 4 sigma from the constraining vertex (low pt step).